src/elbo.jl

```diff
@@ -0,0 +1,81 @@
+using Distributions, LinearAlgebra
+using Bijectors
+using Random
+
+####################################
+# training by minimizing reverse KL
+####################################    
+function elbo_single_sample(
+    x::Union{AbstractVector{T}, T},                          # sample from reference dist q
+    flow::Union{Bijectors.MultivariateTransformed,Bijectors.UnivariateTransformed}  # variational distribution to be trained
+    logp,                                       # lpdf (unnormalized) of the target distribution
+    logq                                        # lpdf (exact) of the reference distribution
+    ) where {T<:Real}
+    y, logabsdetjac = with_logabsdet_jacobian(flow.transform, x)
+    return logp(y) - logq(x) + logabsdetjac
+end
+
+# ELBO based on multiple iid samples
+function elbo(
+    xs::AbstractVector,                         # samples from reference dist q
+    flow::Bijectors.UnivariateTransformed,      # variational distribution to be trained
+    logp,                                       # lpdf (unnormalized) of the target distribution
+    logq                                        # lpdf (exact) of the reference distribution
+    )
+    n_samples = size(xs, 1) # each column is a sample
+    elbo_values = map(x -> elbo_single_sample(x, flow, logp, logq), xs)
+    return sum(elbo_values) / n_samples
```

Comment:
```suggestion
    return mean(elbo_values)
```
is likely have better numerical stability since you don't have to sum a (potentially very large) number before dividing by `n_samples`.